main.go

```diff
@@ -269,9 +273,13 @@ func generate(in configInput) ([]byte, error) {
 			"":    {},
 		}
 		if in.includeDirectPathAuthority {
-			c.Authorities[c2pAuthority] = Authority{
+			authority := Authority{
 				XdsServers: generateServerConfigsFromInputs("dns:///directpath-pa.googleapis.com", in),
 			}
+			if in.includeXDSTPNameInLDS {
+				authority.ClientListenerResourceNameTemplate = fmt.Sprintf("xdstp://%s/envoy.config.listener.v3.Listener/%%s", c2pAuthority)
```

Comment:
I'm surprised to see this for c2p. I had expected the flag to be for plain TD and that c2p would be hard-coded to use xdstp. Looking at `GoogleCloudToProdNameResolver` in Java though, it seems we aren't setting `client_listener_resource_name_template` today. We should discuss what is going on, because I'm missing some context.

Reply:
The c2p resolver in Go doesn't seem to be doing that either.

Reply:
I think you are right Eric, the flag should be for TD. But the authority name that I got from the TD team initially was the c2p authority. But I got it confirmed that  we should be using `traffic-director-global.xds.googleapis.com` for plain TD and for c2p I think we should be overriding this in the client. Updating

Reply:
It seems you are modifying the wrong server configuration. (Or you need to modify both, but with proper authorities)

Reply:
Sorry. I should have taken a second look. Seems to be correct now. 


PS: also realized i was pushing from my master. err...